### ISSUE TITLE
Update system date format for month name abbreviation

### ISSUE THREAD

**Issue by harmeet-status** [sartography/spiff-arena#637]
Switch to the Verbose date format, 21-Oct-2023, when displaying system dates

In markdown instructions, you can now do:

SPIFF_FORMAT:::convert_seconds_to_date_time_for_display(1698677185)

This will display, when rendered on [dev.app](http://dev.app/):

30-10-2023 10:46:25

Once we change the default date format in the dev.app configuration, it will instead display:

30-Oct-2023 10:46:25

I updated https://dev.app.spiff.status.im/process-models/site-administration:grocery to display a date when run.

note that seconds for the current time can be generated in python like: time.time()

**Comment by calexh-sar** [comment]
Once this issue is complete, #698 will be ready to start.

**Comment by jasquat** [comment]
Implemented with #835

**Comment by dinithihj** [comment]
@jasquat If I understand correctly, once we add SPIFFWORKFLOW_FRONTEND_RUNTIME_CONFIG_DATE_FORMAT  to the config, system dates should be displayed according to the specified date format. 

I'm not sure what value should be added there. I tried below but it's not seem to be working.
![image](https://github.com/sartography/spiff-arena/assets/123297840/ec2df607-8dd8-43cd-8a7a-69fb6ae578f1)


Please let me know whether I should do something else to test this,

**Comment by jasquat** [comment]
@dinithihj it's date only. no time. And this is a config that must be set server side. I can put in a PR for the update.

**Comment by jasquat** [comment]
@dinithihj to test this, we will put the code with the "dd-MMM-yyyy" format on sartography and this can be tested there.

After the next release to status-im prod we can change the config.

**Comment by dinithihj** [comment]
@jasquat On sartography environment, When I select a date from the date picker it's giving me an error - Preferred Delivery Date must match format "date"
![image](https://github.com/sartography/spiff-arena/assets/123297840/7c2f51c5-0f5a-46bd-9f59-d2c4f2a499dc)

Process model - https://sartography.spiffworkflow.org/process-models/misc:qa:check-date-format

**Comment by jasquat** [comment]
@dinithihj I removed `"format": "date"` from the jsonschema and now it works. Since the component itself forces the date format the explicit setting of format may not be necessary. Since we do not control that check this may be the best we can do.

@burnettk thoughs?

**Comment by burnettk** [comment]
we put in a fix for this. we want to make sure that the task data has dates in this format: 2024-12-25.

**Comment by dinithihj** [comment]
On sartography environment now the date is displayed in yyyy-mm-dd format.

![image](https://github.com/sartography/spiff-arena/assets/123297840/5b91e858-8ba6-4310-91cf-0acdfe2500a4)
![image](https://github.com/sartography/spiff-arena/assets/123297840/e687e1ff-97dc-404f-b48c-a68cb053c1eb)
![image](https://github.com/sartography/spiff-arena/assets/123297840/1307d48c-c066-4755-9af4-9cb8858bb8be)

**Comment by jasquat** [comment]
Forgot to commit the change to the configs. Updated again and should be good now.

**Comment by dinithihj** [comment]
On Sartography environment date format is displayed as dd-mmm-yyyy on date pickers and process instance view.
![image](https://github.com/sartography/spiff-arena/assets/123297840/f900763e-2386-4559-8899-a725f5d2a006)
![image](https://github.com/sartography/spiff-arena/assets/123297840/074948d8-fabd-49af-947b-b59a48bec222)


However, on task data, it's displaying in yyyy-mm-dd format. And when I printed the date value that also had the yyyy-mm-dd format.
![image](https://github.com/sartography/spiff-arena/assets/123297840/4027dd4a-df69-4b34-b3a9-341a475cc4fc)

**Comment by harmeet-status** [comment]
@burnettk I'm assuming the format in task data needs to stay and we just worry about UI display format?

I think we definitely need to have the date pickers show the date in the correct format @dinithihj

**Comment by jasquat** [comment]
Yes @harmeet-status @dinithihj, we put dates into task data as yyyy-mm-dd so the system can have a consistent date format to deal with. The custom date formats should be for display only.

**Comment by calexh-sar** [comment]
@dinithihj and @jasquat I was checking on the status of this ticket in regards to whether I could start on #698 and noticed that the SPIFFWORKFLOW_FRONTEND_RUNTIME_CONFIG_DATE_FORMAT entry was no longer specified in the Configuration/Secrets.  Am I correct in assuming this was not replaced the last time dev was dumped and perhaps it needs to be included in the Bootstrap process?  

Is there more work needed on this issue or is it safe for me to start on #698 once the Configuration/Secrets is updated?

**Comment by dinithihj** [comment]
@calexh-sar I have tested this on Sartography environment and I don't think we have merged these changes to the dev.app. Also I believe this config is something that needs to be configured on the server side. 

@jasquat @burnettk Please correct me if I'm wrong ^^

**Comment by burnettk** [comment]
Yeah, it’s a Jakub infra change: https://github.com/status-im/infra-spiff-workflow/pull/42. We’ll see what he says.

**Comment by madhurrya** [comment]
@harmeet-status are we going to update the Instructions also to match the same date format? 



![Image](https://github.com/sartography/spiff-arena/assets/110527567/c2d7fa2f-b69d-4336-a73a-b17eb3a5a873)



![Image](https://github.com/sartography/spiff-arena/assets/110527567/d9676bfb-5888-452e-9c23-afd4ec75e540)

**Comment by burnettk** [comment]
it looks like we have a way to convert seconds to a correctly formatted datetime (see https://dev.app.spiff.status.im/process-models/misc:test:dinithi-testing:sartography-issues:559-formatting-in-extension-md), but maybe not a way to convert what i assume is originally "2027-12-31" in the task data to "31 Dec 2027". Though this date formatter without the time thing would be implementable.

**Comment by calexh-sar** [comment]
@madhurrya regarding your question to Harmeet about updating the instructions, that is the intent of issue #698.

**Comment by madhurrya** [comment]
Thanks Alex. I missed that ticket. :-)

**Comment by madhurrya** [comment]
Now we can see the new date format in the system in dev.app and test.app

**Comment by sashayar13** [comment]
@burnettk @jasquat @harmeet-status The change also impacted the Date widgets used in process models (Preffered Delivery Dtae, Start date etc.). Now the field requires a user to enter a date **either in the exact format** or to select it from a calendar.

For example, if a user types 01/04/2024 or 01.04.2024 or 01-apr-2024, the system replaces it with "01-Dec-2024". From a user experience perspective, the field should accept all mentioned inputs and transform them into the "01-Apr-2024".

**Comment by jasquat** [comment]
@sashayar13  #1319 should address most of your concerns. We're attempting to try and interpret what the user is trying to input.

@madhurrya @dinithihj it's out on dev now and ready to test.

**Comment by dinithihj** [comment]
Tested on dev.app (version 369/370) using the model https://dev.app.spiff.status.im/process-models/misc:qa:check-date with different date formats

If I type dates in the formats below, now it's correctly converting them to dd-mmm-yyyy.

- YYYY-MM-DD
- DD.MM.YYYY
- DD/MM/YYYY
- DD/MMM/YYYY
- MMM/DD/YYYY


If I type the date in the formats below, it **WILL NOT** be recognized and converted to dd-mmm-yyyy.

- MM/DD/YYYY
- MM/DD/YY
- YY/MM/DD
- DD/MM/YY
- YYYY/MM/DD
- YYYY/MMM/DD
- YYYY.MM.DD

Sometimes, when I try to erase the date I entered and enter it again, it doesn't convert correctly.
Video Link : https://drive.google.com/file/d/1QV_ydeXu1BoRgJP49SWFBEcERV2r_VGk/view?usp=sharing

**Suggestion** -
If the system can't change the date the user entered to the dd-mmm-yyyy format, I think it's better to leave the date picker value blank instead of filling in an incorrect date. Then the user can return later and enter the correct value. For instance, if I enter 05/30/2020 and then move away from the control, it changes to 05-Dec-2019, which is completely wrong.

**Comment by jasquat** [comment]
We cannot support both mm-dd and dd-mm formats so that makes sense. We don't really support delimiters other than hyphens but we could add some support for periods and slashes but I suggest a new ticket for that if it's desired and keep it out of this ticket since it would be a new feature request. It works in some cases because of the converters we use but we'd need greater effort to make it work in more cases.

I'm not sure what we can do about not displaying any date if it's unrecognized. Unless we check for digits greater than 12 three's really nothing we can do since we don't know what the user's intent is.

I'm not sure what is going on in the video. I will play around a little bit to understand. I think you have to click outside the text field at some point though so it recognizes that something has changed but I'm not sure.

**Comment by dinithihj** [comment]
@sashayar13 Could you please check the list of formats that are supported and not supported? If there are any other formats you need support for, please let us know.

**Comment by sashayar13** [comment]
If I type 

![Image](https://github.com/sartography/spiff-arena/assets/104367255/96b3208e-9027-4764-a0a5-9ddb51eb2002)

The system converts it to

![Image](https://github.com/sartography/spiff-arena/assets/104367255/f0131981-b46f-4b7a-83cd-8f4ce47abde3)

So I think it will be good if we add "DDMMYYYY" format too

**Comment by jasquat** [comment]
@sashayar13 I think any additional date formats should be in a new ticket. This ticket was about month abbreviations in the format and any other formats would have to be estimated and planned separately to figure what we can we cannot support.

**Comment by sashayar13** [comment]
added the ticket https://github.com/sartography/spiff-arena/issues/1420

**Comment by dinithihj** [comment]
Moving this ticket to Ready for Release. 

- Created a new ticket for the random formatting issue when erased and reentered date. https://github.com/sartography/spiff-arena/issues/1423
- The new format request will be covered under the https://github.com/sartography/spiff-arena/issues/1420

**Comment by madhurrya** [comment]
Released to Prod
